src/main/java/emu/grasscutter/server/packet/send/PacketPlayerWorldSceneInfoListNotify.java

```diff
@@ -1,46 +1,46 @@
 package emu.grasscutter.server.packet.send;
 
 import emu.grasscutter.data.GameData;
-import emu.grasscutter.data.excels.scene.SceneTagData;
+import emu.grasscutter.game.player.Player;
 import emu.grasscutter.net.packet.*;
 import emu.grasscutter.net.proto.MapLayerInfoOuterClass;
 import emu.grasscutter.net.proto.PlayerWorldSceneInfoListNotifyOuterClass.PlayerWorldSceneInfoListNotify;
 import emu.grasscutter.net.proto.PlayerWorldSceneInfoOuterClass.PlayerWorldSceneInfo;
-import java.util.stream.IntStream;
+
+import java.util.Map;
 
 public class PacketPlayerWorldSceneInfoListNotify extends BasePacket {
 
-    public PacketPlayerWorldSceneInfoListNotify() {
+    public PacketPlayerWorldSceneInfoListNotify(Player player) {
         super(PacketOpcodes.PlayerWorldSceneInfoListNotify); // Rename opcode later
 
+        var sceneTags = player.getSceneTags();
+
         PlayerWorldSceneInfoListNotify.Builder proto =
                 PlayerWorldSceneInfoListNotify.newBuilder()
-                        .addInfoList(PlayerWorldSceneInfo.newBuilder().setSceneId(1).setIsLocked(false).build())
-                        .addInfoList(
-                                PlayerWorldSceneInfo.newBuilder()
-                                        .setSceneId(3)
+                        .addInfoList(PlayerWorldSceneInfo.newBuilder().setSceneId(1).setIsLocked(false).build());
+
+                        // Iterate over all scenes the user has tags for
+                        for (int scene : sceneTags.keySet()) {
```

Comment:
might not be a good idea to iterate over the scenetags, a scene might not have scenetags, but still need info that needs to be send here.
We probably want to iterate over GameData.getSceneDataMap() filtered by type SCENE_WORLD instead and then get the data we need for them (like the sceneTAgs, and city info from CityConfigData.json)

Reply:
or maybe in the long term something scene specific that also stores unlock/lock could be good instead

Reply:
Some dungeons also use scenetags (Dream dungeons, just as an example) so I wouldn't want to limit it quite that much to just scene_world. A full scene data store implementation might be a bit beyond what I want to add (scenetags vs full scene data), though it would make sense if there's going to be more scene info outside of scenetags and map layers eventually.